ca/transport.go

```diff
@@ -135,6 +135,36 @@ func (c *MutableTLSCreds) LoadNewTLSConfig(newConfig *tls.Config) error {
 	return nil
 }
 
+// UpdateCAs updates the root CAs and client CAs of the existing TLS config in place
+func (c *MutableTLSCreds) UpdateCAs(rootCAs, clientCAs *x509.CertPool) {
+	c.Lock()
+	defer c.Unlock()
+	// TODO(cyli): on upgrade to go 1.8.x, use c.config.Clone() instead
```

Comment:
https://github.com/golang/go/blob/release-branch.go1.7/src/crypto/tls/common.go#L426 seems to only have `Config.clone()`, but https://github.com/golang/go/blob/release-branch.go1.8/src/crypto/tls/common.go#L550 has the exported function (I could just be missing an exported function in 1.7 in a different file, though)